src/server/internalServer.cpp

```diff
@@ -1310,3 +1310,23 @@ std::unique_ptr<Response> InternalServer::handle_locally_customized_resource(con
 }
 
 }
+
+bool ipAvailable(const kiwix::IpAddress& addr, const kiwix::IpMode& mode)
+{
+  bool validV4 = false, validV6 = false;
+  auto interfaces = kiwix::getNetworkInterfacesIPv4Or6();
+
+  if (addr.addr == "127.0.0.1") validV4 = true;
+  if (addr.addr6 == "::1") validV6 = true;
+
+  for (const auto& interface : interfaces) {
+    kiwix::IpAddress interfaceIps = interface.second;
+    validV4 |= !validV4 && interfaceIps.addr == addr.addr;
+    validV6 |= !validV6 && interfaceIps.addr6 == addr.addr6;
+  }
+
+  if (mode == kiwix::IpMode::IPV4) return validV4;
+  else if (mode == kiwix::IpMode::IPV6) return validV6;
+
+  return validV4 && validV6;
+}
```

Comment:
1. Move the definition above the point of usage and hide it in an unnamed namespace
2. Forget about the implementation and describe in a doc comment how you want this function to work, given that `IpAddress` is conceptually a pair of addresses (IPv4 and IPv6). It is intended for the case when only one of `.addr` or `.addr6` is non-empty? If so, then maybe its parameter type should be changed to `std::string`?

Reply:
I agree to change this function's scope into just evaluating whether an ip address (std::string) is available, for the same reasons as we did with `getBestPublicIps`.

I am debating though how the check should be done:
- Mode as parameter (like now) and check the corresponding interface fields.
- No mode as parameter but we check inside the function to determine it's type. Then we check on the corresponding interface fields.
- No mode as parameter and we blindly check both interface fields.

Reply:
I will add a commit implementing the 3rd option which I like most so we can proceed faster later if no disputes.